R/chat.R

```diff
@@ -281,13 +281,24 @@ Chat <- R6::R6Class(
     #'   resolves with the response all at once. Returns a promise that resolves
     #'   to a string (probably Markdown).
     #' @param ... The input to send to the chatbot. Can be strings or images.
-    chat_async = function(...) {
+    #' @param tool_mode Whether tools should be invoked one-at-a-time
+    #'   (`"sequential"`) or concurrently (`"concurrent"`). Sequential mode is
+    #'   best for interactive applications, especially when a tool may involve
+    #'   an interactive user interface. Concurrent mode is best for automated
+    #'   scripts or non-interactive applications.
+    chat_async = function(..., tool_mode = c("sequential", "concurrent")) {
```

Comment:
Do you think we should default to concurrent or sequential tool calling in the async chat/stream methods?

* concurrent: status quo, makes sense for non-interactive use cases
* sequential: best when the chat is powering a user interface, e.g. shinychat

Ultimately it's unlikely to matter much for most tools unless they're async or correctly set up to use true parallel processing. `"sequential"` seems like a good default, but if you've gone through the trouble of setting up `future` or `mirai`, it'd be nicer for async tool calling to _just work_ than to also need to change this argument value

Reply:
Go with your gut. I think shinychat is going to be 99% of the users of this function 😄

Reply:
Thanks! Okay, my gut says `"concurrent"`. `"sequential"` is the weirder use-case, and the distinction only becomes relevant at the most extreme end of the async spectrum: async streaming with async tool functions that do actual async work.